day_7/ceasar_cipher.rb

```diff
@@ -0,0 +1,36 @@
+class CeasarCipher
+  def encode(text, shift)
+    text = text.chars.to_a
+    text_index = []
+    plain = [*"A".."Z"]
+    text.each do |t|
+      t = t.upcase
+      if plain.include?(t)
+        text_index << plain.index(t).to_i
+      else
+        text_index << t
+      end
+    end
+
+    cipher = [*"A".."Z"]
+    shift.times do |x|
+      x = cipher.shift
+      cipher << x
+    end
```

Comment:
I'm not sure this is accomplishing a shifted alphabet in the way you think. Check out ruby's `rotate` method to see if it's shifting your alphabet more consistently.

As a general rule, you'll want to stay away from naming variables the same as ruby methods `shift` in this case.